### PR TITLE
Story1

### PR DESCRIPTION
**Pull request by JamesCullen2**
Hi Charlie/Mike, 

Please find pull request for completed work; database and CMS form with links between form and db. 

Many thanks, 

James

### CODE REVIEW

File: admin.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
+<?php
+/**this function is to get subtitle column data from the about_me table in the database
```
Comment (charliecog) [inline admin.php]: don't need to refer to 'this function'

File: admin.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * @param $db array data to be selected from
+ *
+ * @return array of subtitle column values
+ */
+function getAboutMeSubtitleFromDatabase($db) {
```
Comment (charliecog) [inline admin.php]: return type hinting

File: admin.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    $query = $db->prepare("SELECT `id`,`subtitle` FROM `about_me` WHERE `deleted` = 0;");
+    $query->execute();
+    return $query->fetchAll();
+}
+/**this function is to return the values of each subtitle in about_me section into a drop-down form
```
Comment (charliecog) [inline admin.php]: pop a line break inbetween functions (do for all of them)

File: admin.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * @param $subtitleArray array of all values in subtitle field from about_me table
+ *
+ * @return string of all subtitles
+ */
+function listingSubtitles($subtitleArray) {
```
Comment (charliecog) [inline admin.php]: return type hinting

File: admin.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    $results = " ";
+    foreach ($subtitleArray as $value) {
+        $results .=  "<option value='" . $value['id'] . "'>" . $value['subtitle'] . "</option>";
+    }
+    return $results;
+}
+/**this function is to get text column from about_me table in db
+ *
+ * @param $db array data to be selected from
+ *
```
Comment (charliecog) [inline admin.php]: dont need line break here

File: admin.php [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+ * @param $id mixed data to be selected from
+ *
+ * @return mixed text field from about_me table
+ */
+function getAboutMeDescriptionFromDatabase($db, $id) {
```
Comment (charliecog) [inline admin.php]: type hinting and return type hinting

File: admin.php [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    $query->execute();
+    return $query->fetch();
+}
+/**this function is to return text field value of about_me section
+ *
+ * @param $textArray array of text field from about me table
+ *
+ * @return string text field from about me table
+ */
+function populateDescriptionForm($textArray) {
```
Comment (charliecog) [inline admin.php]: type hinting and return type hinting

File: admin.php [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * @param $db array of subtitles and text to be sent to db
+ *
+ * @param $newSubtitle string subtitle to be sent to db
+ *
+ * @param $newDescription string text description to be sent to db
+ *
+ * @return  string of subtitle and text values to populate new row of about_me table
+ */
+function addAboutMe($db, $newSubtitle, $newDescription) {
```
Comment (charliecog) [inline admin.php]: type hinting and return type hinting

File: admin.php [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * @return string text field from about me table
+ */
+function populateDescriptionForm($textArray) {
+    return $textArray['text'];
+}
+/**this function is to edit current data with new data
+ *
+ * @param $db array data to be selected from
+ *
```
Comment (charliecog) [inline admin.php]: dont need line break

File: admin.php [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+ * @param $oldText string current data
+ *
```
Comment (charliecog) [inline admin.php]: dont need line break

File: admin.php [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+function updateDescription($db, $oldText, $newText) {
+    $query = $db->prepare("UPDATE `about_me` SET `text` = :newText WHERE `text` = :oldText;");
+    $query->bindParam(':oldText', $oldText);
+    $query->bindParam(':newText', $newText);
+    $query->execute();
+}
+/**this function is to insert new data into about_me table in db
+ *
+ * @param $db array of subtitles and text to be sent to db
+ *
```
Comment (charliecog) [inline admin.php]: dont need line break

File: admin.php [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+ * @param $newSubtitle string subtitle to be sent to db
+ *
```
Comment (charliecog) [inline admin.php]: dont need line break

File: admin.php [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+    $query = $db->prepare("INSERT INTO `about_me` (`subtitle`,`text`)
+                            VALUES (:newSubtitle,:newDescription);");
+    $query->bindParam(':newSubtitle', $newSubtitle);
+    $query->bindParam(':newDescription', $newDescription);
+    $query->execute();
+}
+/**this function is to delete data from about_me table
+ *
+ * @param $db array of subtitle and text from db
+ *
```
Comment (charliecog) [inline admin.php]: dont need line break

**Review by charliecog — CHANGES_REQUESTED** [review]
Like it, good clean code, well organised and doc blocked. prefer this to having loads of wild code that is hard to interpret. couple of quick changes here + the stuff we talked about.

File: admin.php (hunk 1 above)
Comment (JamesCullen2) [reply admin.php]: all edited

File: admin.php (hunk 2 above)
Comment (JamesCullen2) [reply admin.php]: done

File: admin.php (hunk 3 above)
Comment (JamesCullen2) [reply admin.php]: cool, all done

File: admin.php (hunk 5 above)
Comment (JamesCullen2) [reply admin.php]: cool, taken it out

File: admin.php (hunk 9 above)
Comment (JamesCullen2) [reply admin.php]: taken out

File: admin.php (hunk 10 above)
Comment (JamesCullen2) [reply admin.php]: done

File: admin.php (hunk 11 above)
Comment (JamesCullen2) [reply admin.php]: done

File: admin.php (hunk 12 above)
Comment (JamesCullen2) [reply admin.php]: done

File: admin.php (hunk 13 above)
Comment (JamesCullen2) [reply admin.php]: done

File: admin.php (hunk 4 above)
Comment (JamesCullen2) [reply admin.php]: done

File: admin.php (hunk 6 above)
Comment (JamesCullen2) [reply admin.php]: done

File: admin.php (hunk 7 above)
Comment (JamesCullen2) [reply admin.php]: done

**Pull request closed without merging**